test/libsolidity/smtCheckerTests/operators/delete_function_type_1.sol

```diff
@@ -12,5 +12,7 @@ contract C {
     }
 }
 // ----
+// Warning 3075: (128-136): Comparison of internal function pointers can yield unexpected results in the legacy pipeline with the optimizer enabled, and will be disallowed entirely in the next breaking release.
 // Warning 7229: (128-136): Assertion checker does not yet implement the type function () for comparisons
 // Warning 6328: (121-137): CHC: Assertion violation happens here.
```

Comment:
The assert in this and other SMTChecker tests probably need to be changed to something else. If we're telling people not to rely on this comparison working, we should not rely on it ourselves either :)

For example this one looks like it does not really need the function to be internal, because it's just testing `delete` so maybe an external one would be fine here.

Reply:
Don't think I've ever touched SMT tests before - where would I do this?

Reply:
The message comes from `assert(s.e == f)` above. If you change both function pointer declarations from `function()` to `function() external`, the assert should still pass, but without a warning about internal function comparison. Then we won't have to remove this test after we disallow these comparisons.

Similar in other tests. The only one we can't adjust that way is `function_type_to_function_type_internal.sol`. There we actually do want an internal function. You can leave that one as is, we'll remove it once we disallow comparisons completely. And no need to add an external version because it already has one.